Forgot Password not working

@machristie   If I type in my user name on the “Request Password Reset” subform and click “REQUEST PASSWORD RESET” button, nothing happens other than a quick glimpse of a load bar on the subform. Then when I went to login with my old/normal password, it worked just fine so apparently my password was not reset.

In the Apache logs I see these errors:

>File "/www/htdocs/test/whispersservices/whispersapi/views.py", line 2177, in reset_password
>serializer = self.get_serializer_class()(user, data=data, partial=True, context={'request': request})
>AssertionError: 'UserViewSet' should either include a `serializer_class` attribute, or override the `get_serializer_class()` method.

Which is a little strange, because that ViewSet definitely has a serializer class set (line 2074):
```serializer_class = UserSerializer```

@machristie perhaps it has something to do with the extra parentheses on line 2177 (shown in excerpt above).

--- a/src/app/reset-password/reset-password.component.html
+++ b/src/app/reset-password/reset-password.component.html
@@ -17,7 +17,7 @@
 
 <mat-dialog-actions>
   <div class="form-buttons-flex">
-    <button mat-button (click)="this.resetPasswordDialogRef.close('cancel');">
+    <button mat-stroked-button (click)="this.resetPasswordDialogRef.close('cancel');">
       Cancel
     </button>
     <button mat-raised-button color="primary" type="submit" [disabled]="resetPasswordForm.invalid"

--- a/src/app/reset-password/reset-password.component.ts
+++ b/src/app/reset-password/reset-password.component.ts
@@ -80,7 +80,12 @@ export class ResetPasswordComponent implements OnInit {
 
           let errorMessage = error;
           try {
-            errorMessage = JSON.parse(error).status;
+            const parsedResponse = JSON.parse(error);
+            if ('status' in parsedResponse) {
+              errorMessage = parsedResponse.status;
+            } else if ('non_field_errors' in parsedResponse) {
+              errorMessage = parsedResponse.non_field_errors.join(", ");
+            }
           } catch (error) {
             // Ignore JSON parsing error
           }

